### PR TITLE
Smooth slider behavior.

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Update graphs immediately, and debounce layer filter updates.

### CODE REVIEW

File: js/OTPALayer.js [hunk 1]
```diff
@@ -49,32 +49,35 @@ L.OTPALayer = L.FeatureGroup.extend({
...
-                  }
-                  layer.setStyle(self._highlightedPointsetStyle());
-                  layer.bringToFront();
-
-                  self.fireEvent('select', {data: layer.feature.properties});
-              },
-
-              mouseout: function resetHighlight(e) {
-                  lastLayer = layer;
+      return;
```
Comment (maurizi) [inline js/OTPALayer.js]: I'm pretty sure this return statement will break showing point information...

File: index.html [hunk 2]
```diff
@@ -12,7 +12,6 @@
   <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.2.0/jquery.min.js"></script>
   <script src="https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.0.0/lodash.min.js"></script>
   <script src="js/d3.min.js" charset="utf-8"></script>
-  <script src="js/leaflet-pip.js" charset="utf-8"></script>
```
Comment (maurizi) [inline index.html:15]: oops... good catch

File: js/OTPALayer.js (hunk 1 above)
Comment (flibbertigibbet) [inline js/OTPALayer.js]: Whoops, didn't mean to commit that. Good catch. (Was trying to find a way to make the draggable marker not be super slow; seems to be related to [this](https://github.com/Leaflet/Leaflet/issues/2164).)

**Comment by maurizi** [comment]
So I wasn't encountering issues with filtered points appearing on page load, or when moving the marker.

Its a bit convoluted, but `_createSurface` calls `_updateIsochronesIndicators` (possibly in the callback of `_getPointset`), which calls `_getIndicator`, which calls `_showFilteredPointset` if `self._pointset` is defined.  I think with your changes `_showFilteredPointset` will be called twice moving the marker or on page load.

**Comment by flibbertigibbet** [comment]
I was seeing `self._pointsetData` sometimes not yet being defined on page load or marker move when the filter method was called, due to timing issues; there were a series of AJAX calls being fired off one after another without checking to see if a preceding dependent call had finished. The filter was getting called twice on marker move (fixed) but once on page load.

**Comment by maurizi** [comment]
:+1: Playing with the slider is so much snappier now.

**Pull request merged**